xgcm/grid_ufunc.py

```diff
@@ -475,54 +477,37 @@ def apply_as_grid_ufunc(
         raise ValueError(
             "To apply a boundary condition you must provide the widths of the boundaries"
         )
+
     if boundary_width:
         # convert dummy axes names in boundary_width to match real names of given axes
         boundary_width_real_axes = {
             dummy_to_real_axes_mapping[ax]: width
             for ax, width in boundary_width.items()
         }
-
-        padded_args = grid.pad(
-            *args,
-            boundary_width=boundary_width_real_axes,
-            boundary=boundary,
-            fill_value=fill_value,
-        )
     else:
         # If the boundary_width kwarg was not specified assume that zero padding is required
         boundary_width_real_axes = {
             real_ax: (0, 0) for real_ax in dummy_to_real_axes_mapping.values()
         }
-        padded_args = args
 
-    if any(
-        _has_chunked_core_dims(padded_arg, core_dims)
-        for padded_arg, core_dims in zip(padded_args, in_core_dims)
-    ):
-        # merge any lonely chunks on either end created by padding
-        rechunked_padded_args = _rechunk_to_merge_in_boundary_chunks(
-            padded_args,
-            args,
-            boundary_width_real_axes,
-            grid,
-        )
-    else:
-        rechunked_padded_args = padded_args
+    # Compose the ufunc with the padding needed to replace trimmed elements
+    padder = Padder(grid, boundary_width_real_axes, boundary, fill_value)
+    pad_then_func = padder.compose_with(func)
```

Comment:
Means that this is all the reference to padding I need to have in `apply_as_grid_ufunc`